app/Services/ModuleService.php

```diff
@@ -70,4 +83,203 @@ public function getAdminLinks(): array
+     * to install module in the Storage.
+     *
+     * @param UploadedFile $file
+     *
+     * @return bool
+     */
+    public function installModule(UploadedFile $file): bool
+    {
+        $file_ext = $file->getClientOriginalExtension();
+        $allowed_extensions = ['zip', 'tar', 'gz'];
+
+        if (!in_array($file_ext, $allowed_extensions, true)) {
+            return false;
+        }
+
+        $module = null;
+
+        $new_dir = File::makeDirectory(uniqid());
+        $temp_ext_folder = storage_path('app/tmp/modules/'.$new_dir);
+
+        $temp = $temp_ext_folder;
+
+        $zipper = null;
+
+        if ($file_ext === 'tar' || $file_ext === 'gz') {
+            $zipper = new PharData($file);
+            $zipper->decompress();
+        }
+
+        if ($file_ext === 'zip') {
+            $madZipper = new Madzipper();
+
+            try {
+                $zipper = $madZipper->make($file);
+            } catch (Exception $e) {
+                Log::emergency('Could not extract zip file.');
+            }
+        }
+
+        try {
+            $zipper->extractTo($temp);
+        } catch (Exception $e) {
+            Log::emergency('Cannot Extract Module!');
+        }
+
+        if (!File::exists($temp.'/module.json')) {
+            $directories = Storage::directories('tmp/modules/'.$new_dir);
+            $temp = storage_path('app').'/'.$directories[0];
+        }
+
+        $json_file = $temp.'/module.json';
+
+        if (File::exists($json_file)) {
+            $json = json_decode(file_get_contents($json_file), true);
+            $module = $json['name'];
+        } else {
+            return false;
+        }
+
+        if (!$module) {
+            return false;
+        }
+
+        $toCopy = base_path().'/modules/'.$module;
+
+        if (File::exists($toCopy)) {
+            throw new ModuleExistsException($module);
+        }
+
+        File::moveDirectory($temp, $toCopy);
+
+        File::deleteDirectory($temp_ext_folder);
+
+        $this->addModule($module);
+
+        Artisan::call('config:cache');
+        Artisan::call('module:migrate  '.$module);
+        return true;
+    }
+
+    /**
+     * Update module with the status passed by user.
+     *
+     * @param $id
+     * @param $status
+     *
+     * @return bool
+     */
+    public function updateModule($id, $status): bool
+    {
+        $module = Module::find($id);
+        $module->update([
+            'enabled' => $status,
+        ]);
+        return true;
+    }
+
+    /**
+     * Delete Module from the Storage & Database.
+     *
+     * @param $id
+     * @param $data
+     *
+     * @return bool
+     */
+    public function deleteModule($id, $data): bool
+    {
+        $module = Module::find($id);
+        if ($data['verify'] === strtoupper($module->name)) {
+            try {
+                $module->delete();
+            } catch (Exception $e) {
+                Log::emergency('Cannot Delete Module!');
```

Comment:
throw and error/return here, this just logs it and continues on with trying to delete